test/OrchardCore.Tests/OrchardCore.Users/AccountControllerTests.cs

```diff
@@ -7,13 +8,164 @@
 using OrchardCore.Tests.Apis.Context;
 using OrchardCore.Users;
 using OrchardCore.Users.Controllers;
+using OrchardCore.Users.Handlers;
+using OrchardCore.Users.Indexes;
 using OrchardCore.Users.Models;
 using OrchardCore.Users.ViewModels;
 
 namespace OrchardCore.Tests.OrchardCore.Users;
 
-public class RegistrationControllerTests
+public class AccountControllerTests
 {
+    [Fact]
+    public async Task ExternalLoginSignIn_Test()
+    {
+        // Arrange
+        var context = await GetSiteContextAsync(new RegistrationSettings()
+        {
+            UsersCanRegister = UserRegistrationType.AllowRegistration,
+        });
+
+        var responseFromGet = await context.Client.GetAsync("Register");
+
+        Assert.True(responseFromGet.IsSuccessStatusCode);
+
+        // Act
+        var model = new RegisterViewModel()
+        {
+            UserName = "TestUserName",
+            Email = "test@orchardcore.com",
+            Password = "test@OC!123",
+            ConfirmPassword = "test@OC!123",
+        };
+
+        var response = await context.Client.SendAsync(await CreateRequestMessageAsync(model, responseFromGet));
+
+        // Assert
+        Assert.Equal(HttpStatusCode.Redirect, response.StatusCode);
+        Assert.Equal($"/{context.TenantName}/", response.Headers.Location.ToString());
+     
+        await context.UsingTenantScopeAsync(async scope =>
+        {
+            var userManager = scope.ServiceProvider.GetRequiredService<UserManager<IUser>>();
+
+            var user = await userManager.FindByNameAsync(model.UserName) as User;
+
+            var externalClaims = new List<SerializableClaim>();
+            var userRoles = await userManager.GetRolesAsync(user);
+
+            var context = new UpdateUserContext(user, "TestLoginProvider", externalClaims, userRoles);
+            var scriptExternalLoginEventHandler = scope.ServiceProvider.GetServices<IExternalLoginEventHandler>()
+                        .FirstOrDefault(x => x.GetType() == typeof(ScriptExternalLoginEventHandler)) as ScriptExternalLoginEventHandler;
+            var loginSettings = new LoginSettings
+            {
+                UseScriptToSyncRoles = true,
+                SyncRolesScript = """
+                context.claimsToUpdate.push({claimType:"displayName", claimValue:"Sam Zhang(CEO)"});
+                context.claimsToUpdate.push({claimType:"firstName",   claimValue:"Sam"});
+                context.claimsToUpdate.push({claimType:"lastName",    claimValue:"Zhang"});
+                context.claimsToUpdate.push({claimType:"jobTitle",    claimValue:"CEO"});
+                context.rolesToAdd.push("Administrator")
+                context.propertiesToUpdate={
```

Comment:
Hi @gvkries ，Do you think this unit test will cover your usage scenario?

Reply:
Yes, thank you.